3scaleAdapter/scripts/generate-config.sh

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env bash
+
+# This script provides a way to generate the required files for an out-of-process adapter
+# Requires Istio source code to be on $GOPATH and will generate based on the locally checked
+# out version of Istio
+# Runs a diff of generated files against current configuration
+
+script_dir=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+root_dir=$(dirname "${script_dir}")
+
+istio_src="${GOPATH-$HOME/go}/src/istio.io/istio"
+mixer_src="${istio_src}/mixer"
+target_dir="${mixer_src}/adapter/3scaleAdapter"
+
+
+# Test the existence of provided directory
+# Args: (path to directory)
+verify_dir_exists() {
+    if [[ ! -d "${1}" ]];
+    then
+        echo "Required directory not found in ${1}"
+        exit 1
+    fi
+}
+
+# Recursive copy of a given directory to a target directory
+# Args: (source directory, target directory)
+copy_files() {
+    cp -r "${1}" "${2}"
+}
+
+# Copies the required files in place for go generate command
+copy_adapter_files() {
+    verify_dir_exists "${root_dir}/config"
+    copy_files pkg/threescale "${target_dir}"
+    copy_files config/config.proto "${target_dir}/config"
+}
+
+verify_dir_exists "${istio_src}"
+mkdir -p "${target_dir}/config"
+copy_adapter_files
+cd "${target_dir}" && go generate ./...
+
+compare_extensions=(
+  yaml
+  go
+  proto_descriptor
+)
+
+for i in "${compare_extensions[@]}" ; do
+  for file in "${target_dir}/config/"*.${i} ; do
+    diff -b "$file" "config/${file##*/}"
+  done
+done
```

Comment:
two things to improve here (we can leave them for another PR):
* we can check whether the diff generated any output (ie. there are diffs) and apart from printing it out we can also exit with a value of success (0) if no diff or error (1 or other values) if there was any diff.
* we can wrap the code from `verify_dir_exists` to the `for` loop into a `main` function, then below compare whether the script is being executed or sourced: see https://github.com/unleashed/rbenv4ci/blob/fec285280938b2f0a7e9810f5d3e9c96c34f25cf/bin/bundle_install_rubies#L32-L37 (instead of main there is another function being invoked, but the point remains).